src/model/kubernetes.js

```diff
@@ -0,0 +1,329 @@
+const KubeClient = require('kubernetes-client').Client;
+const core = require('@actions/core');
+const base64 = require('base-64');
+
+class Kubernetes {
+  static async runBuildJob(buildParameters, baseImage) {
+    // uses default kubeconfig location/env variable
+    const kubeClient = new KubeClient();
+    await kubeClient.loadSpec();
+
+    const buildId = Kubernetes.uuidv4();
+    const pvcName = `unity-builder-pvc-${buildId}`;
+    const secretName = `build-credentials-${buildId}`;
+    const jobName = `unity-builder-job-${buildId}`;
+
+    Object.assign(this, {
+      kubeClient,
+      buildId,
+      buildParameters,
+      baseImage,
+      pvcName,
+      secretName,
+      jobName,
+    });
+
+    await Kubernetes.createSecret();
+    await Kubernetes.createPersistentVolumeClaim();
+    await Kubernetes.createBuildJob();
+    await Kubernetes.watchBuildJobUntilFinished();
+    await Kubernetes.cleanupJob();
+
+    core.setOutput('kubernetesPVC', pvcName);
+  }
+
+  static async createSecret() {
+    const secretManifest = {
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        name: this.secretName,
+      },
+      type: 'Opaque',
+      data: {
+        GITHUB_TOKEN: base64.encode(this.buildParameters.githubToken),
+        UNITY_LICENSE: base64.encode(process.env.UNITY_LICENSE),
+        ANDROID_KEYSTORE_BASE64: base64.encode(this.buildParameters.androidKeystoreBase64),
+        ANDROID_KEYSTORE_PASS: base64.encode(this.buildParameters.androidKeystorePass),
+        ANDROID_KEYALIAS_PASS: base64.encode(this.buildParameters.androidKeyaliasPass),
+      },
+    };
+    await this.kubeClient.api.v1.namespaces('default').secrets.post({ body: secretManifest });
+  }
+
+  static async createPersistentVolumeClaim() {
+    const pvcManifest = {
+      apiVersion: 'v1',
+      kind: 'PersistentVolumeClaim',
+      metadata: {
+        name: this.pvcName,
+      },
+      spec: {
+        accessModes: ['ReadWriteOnce'],
+        volumeMode: 'Filesystem',
+        resources: {
+          requests: {
+            storage: '5Gi',
+          },
+        },
+      },
+    };
+    await this.kubeClient.api.v1
+      .namespaces('default')
+      .persistentvolumeclaims.post({ body: pvcManifest });
+    let ready = false;
+    while (!ready) {
+      // eslint-disable-next-line no-await-in-loop
+      const queryResult = await this.kubeClient.api.v1
+        .namespaces('default')
+        .persistentvolumeclaims(this.pvcName)
+        .get();
+      ready = queryResult.body.status.phase !== 'Pending';
+    }
+    core.info('PVC created');
```

Comment:
Also here I would go `Persistent Volume claimed` or something else that's descriptive and helps developers debug their builds when something breaks.

Reply:
I went with "Persistent Volume ready for claims", that would be more accurate here.